src/connection_string.ts

```diff
@@ -291,9 +294,17 @@ export function parseOptions(
   }
 
   for (const key of url.searchParams.keys()) {
-    const values = [...url.searchParams.getAll(key)];
+    const values = url.searchParams.getAll(key);
+
+    const isReadPreferenceTags = /readPreferenceTags/i.test(key);
+
+    if (!isReadPreferenceTags && values.length > 1) {
+      throw new MongoInvalidArgumentError(
+        `URI option "${key}" cannot appear more than once in the connection string`
+      );
+    }
 
-    if (values.includes('')) {
+    if (!isReadPreferenceTags && values.includes('')) {
       throw new MongoAPIError('URI cannot contain options with no value');
```

Comment:
I think for being a bit more helpful to users we should provide the `key` in the error message to be more clear which option had the empty value.